### ISSUE TITLE
Ability to get current TraceId + SpanId for context propagation

### ISSUE THREAD

**Issue by cfraenkel** [craigatk/opentelemetry-gradle-plugin#191]
I'm calling tools that support OTEL tracing from my gradle tasks via exec. One example for this is `docker build` (using the buildx plugin). I would like to set the TRACEPARENT environment variable for these tasks correctly so that traces from these tools are associated with the current task and show up in Jaeger correctly

I've tried to to use `Context.current()` for this but it didn't return the current context or span from the task listener.

```kotlin
  tasks.register("buildImage") {
//...
      exec {
        W3CTraceContextPropagator.getInstance()
          .inject(Context.current(), this) { execSpec: ExecSpec?, key: String, value: String ->
            execSpec?.environment(key.uppercase(), value)
          }
        commandLine(listOf("docker", "buildx", "build", "."))
      }
}
```

**Comment by craigatk** [comment]
Thanks for reporting this! I'll dig into it and see what potential solutions I can come up with.

**Comment by craigatk** [comment]
I added the ability to pass the span and trace IDs to all `Exec` tasks in the build in version 3.0.0 of the plugin. Docs on it are here: https://github.com/craigatk/opentelemetry-gradle-plugin?tab=readme-ov-file#pass-trace-and-span-ids-to-exec-tasks-as-environment-variables

Please give it a try and let me know if it works for you!

**Comment by cfraenkel** [comment]
thanks, I've tried it out:

```kotlin
// ...
val jaegerUrl = "http://localhost:4317"

openTelemetryBuild {
  endpoint.set(jaegerUrl)
  taskTraceEnvironmentEnabled.set(true)
}

val runDocker by tasks.registering(Exec::class) {
  commandLine("docker", "buildx", "build", ".")
  environment("OTEL_EXPORTER_OTLP_ENDPOINT", jaegerUrl)
  doFirst {
    val traceId = environment["TRACE_ID"]
    val spanId = environment["SPAN_ID"]
    environment("TRACEPARENT", "00-$traceId-$spanId-01")
  }
}
```

I've uploaded a full reproducer at https://github.com/cfraenkel/gradle-otel-test/blob/main/build.gradle.kts

Following points:
* I currently have to do a doFirst and define the w3c `TRACEPARENT` header myself that expected by the build tool.
* I think the SPAN_ID in the exec task is the wrong one - it looks like its the span ID of the build instead of the exec task. The two spans are siblings instead of nested:
![image](https://github.com/user-attachments/assets/0d2382f9-2f1b-4b52-bb11-8108760dff82)

**Comment by craigatk** [comment]
Thanks for the example and details on those issues! I'll investigate 👀

**Comment by craigatk** [comment]
I added the `TRACEPARENT` environment variable and fixed the exec task parent association in `3.1.0` - please give it a try!

**Comment by craigatk** [comment]
Awesome, glad it's working! :)
